servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -513,13 +513,15 @@ private LogEntry logEntryOperationsAccessCheck(
 
     Set<Check> checks = newHashSetWithExpectedSize(operations.size());
     for (Operation op : operations) {
-      if (op instanceof Operation.Put) {
-        checks.add(canReadContentKey(endRef.getValue(), identifiedKeys.get(op.getKey())));
-      } else if (op instanceof Operation.Delete) {
-        checks.add(canReadContentKey(endRef.getValue(), identifiedKeys.get(op.getKey())));
-      } else {
+      if (!(op instanceof Operation.Put) && !(op instanceof Operation.Delete)) {
         throw new IllegalStateException("Unknown operation " + op);
       }
+      IdentifiedContentKey identifiedKey = identifiedKeys.get(op.getKey());
+      if (identifiedKey != null) {
```

Comment:
I believe we actually have content IDs for `Delete` ops in the `StoreIndex` in the associated commit. I guess `VersionStore.getIdentifiedKeys()` should be able to discover keys for all operations. WDYT?

Reply:
I thought about that too, but everything indicates that `getIdentifiedKeys` was designed to only handle PUT operations. The existing unit test does not test DELETE at all.

Let me play a bit with the idea though.

Reply:
OTOH, it seems that `getIdentifiedKeys` is only used in `logEntryOperationsAccessCheck`, so I think we can make it behave the way we want without introducing regressions.

Reply:
Missing delete tests might be an oversight :thinking: I think if we can recover deleted IDs in that method, we should :)

Reply:
I changed `VersionStore` as you suggested. I kept the changes in this method just in case. Does that sound OK to you?

Reply:
Yeah, keeping this change looks reasonable to me :+1: